src/main/java/teammates/logic/core/StudentsLogic.java

```diff
@@ -197,23 +197,20 @@ public StudentAttributes updateStudentCascade(StudentAttributes.UpdateOptions up
 
     public List<StudentAttributes> updateStudentCascadeBatch(List<StudentAttributes.UpdateOptions> updateOptionsList)
             throws CascadingTransactionException {
-        StudentsDb.BatchUpdateStudentsTransaction transaction =
+        StudentsDb.BatchUpdateStudentsTransaction studentUpdateTransaction =
                 new StudentsDb.BatchUpdateStudentsTransaction(studentsDb, updateOptionsList);
-        List<StudentUpdate> studentUpdates = transaction.getUpdatedStudents();
+        List<StudentUpdate> studentUpdates = studentUpdateTransaction.getUpdatedStudents();
 
-        // TODO: add in update response for changing email/team/section
         CascadingTransaction responseUpdateForChangingEmailTransaction =
                 frLogic.updateFeedbackResponsesForChangingEmailBatch(studentUpdates);
-        responseUpdateForChangingEmailTransaction.withUpstreamTransaction(transaction);
-
-        // TODO: batch deletion
-        frLogic.updateFeedbackResponsesForChangingTeamBatch(studentUpdates);
-
+        frLogic.updateFeedbackResponsesForChangingTeamBatch(studentUpdates); // TODO: batch deletion
```

Comment:
Should we move the updating logic into CascadingTransaction class? That way we can carry out a transaction by just calling ```CascadingTransaction().execute()``` or smth similar

Reply:
This is deletion, which I have not made into a cascading transaction. It is part of a todo I guess but I’d expect it to be simpler and hence not urgent to add transaction interface.